npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater/npm_lockfile_updater_spec.rb

```diff
@@ -197,202 +142,258 @@
+          expect { updated_npm_lock_content }.
+            to raise_error(Dependabot::DependencyFileNotResolvable)
+        end
       end
-    end
 
-    context "when sub dependency version is missing" do
-      let(:files) { project_dependency_files("npm6/github_sub_dependency_version_missing") }
+      context "with an invalid requirement in the package.json" do
+        let(:files) { project_dependency_files("#{npm_version}/invalid_requirement") }
 
-      let(:dependency_name) { "test-missing-dep-version-npm-package" }
-      let(:requirements) do
-        [{
-          requirement: nil,
-          file: "package.json",
-          groups: ["dependencies"],
-          source: {
-            type: "git",
-            url: "https://github.com/dependabot-fixtures/"\
-            "test-missing-dep-version-npm-package",
-            branch: nil,
-            ref: ref
-          }
-        }]
+        let(:npm_lock_fixture_name) { "package-lock.json" }
+
+        it "raises a helpful error" do
+          expect { updated_npm_lock_content }.
+            to raise_error(Dependabot::DependencyFileNotResolvable)
+        end
       end
-      let(:previous_requirements) do
-        [{
-          requirement: nil,
-          file: "package.json",
-          groups: ["dependencies"],
-          source: {
-            type: "git",
-            url: "https://github.com/dependabot-fixtures/"\
-            "test-missing-dep-version-npm-package",
-            branch: nil,
-            ref: old_ref
-          }
-        }]
+
+      context "because we're updating to a nonexistent version" do
+        let(:files) { project_dependency_files("#{npm_version}/simple") }
+
+        let(:dependency_name) { "fetch-factory" }
+        let(:version) { "5.0.2" }
+        let(:requirements) do
+          [{
+            file: "package.json",
+            requirement: "^5.0.2",
+            groups: ["dependencies"],
+            source: nil
+          }]
+        end
+
+        it "raises an unhandled error" do
+          expect { updated_npm_lock_content }.
+            to raise_error(Dependabot::InconsistentRegistryResponse)
+        end
       end
-      let(:previous_version) { "f56186c1643a9a09a86dfe09b1890921330c28bb" }
-      let(:version) { "3deb2768be1591f2fdbdec40aa6a63ba6b270b40" }
-      let(:ref) { "v1.0.3" }
-      let(:old_ref) { "v1.0.2" }
 
-      it "raises an error" do
-        expect { updated_npm_lock_content }.
-          to raise_error(Dependabot::SharedHelpers::HelperSubprocessFailed)
+      context "with a dependency that can't be found" do
+        let(:files) { project_dependency_files("#{npm_version}/nonexistent_dependency_yanked_version") }
+
+        it "raises a helpful error" do
+          expect { updated_npm_lock_content }.
+            to raise_error(Dependabot::PrivateSourceAuthenticationFailure)
+        end
       end
-    end
 
-    context "when a git src dependency doesn't have a valid package.json" do
```

Comment:
This was only removed from npm 7 as it takes absolutely ages and actually manages to do the update, the spec exists here for npm 6: https://github.com/dependabot/dependabot-core/blob/bfbc137cddb2da65f0e37f6ee4fae8c6dee75067/npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater/npm_lockfile_updater_spec.rb#L79